fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanAccrualsProcessingServiceImpl.java

```diff
@@ -318,7 +318,9 @@ private void addAccruals(@NotNull Loan loan, @NotNull LocalDate tillDate, boolea
             }
         }
 
-        AccrualPeriodsData accrualPeriods = calculateAccrualAmounts(loan, tillDate, periodic);
+        LocalDate calculationTillDate = loan.isProgressiveSchedule()
```

Comment:
I am afraid we need to limit the till date amendment only for calculating the interest. We dont want to pick up any charges early...